### PR TITLE
pkg/manifests: add handling tolerations in components config

### PR DESCRIPTION
**Pull request by paulfantom**

### CODE REVIEW

File: pkg/manifests/manifests.go [hunk 1]
```diff
@@ -290,6 +290,10 @@ func (f *Factory) AlertmanagerMain(host string) (*monv1.Alertmanager, error) {
 		a.Spec.NodeSelector = f.config.AlertmanagerMainConfig.NodeSelector
 	}
 
+	if f.config.AlertmanagerMainConfig.Tolerations != nil {
```
Comment (brancz) [inline pkg/manifests/manifests.go]: best is to check `len(tolerations)` in each of these as in go a nil slice will also return length 0

File: assets/prometheus-k8s/prometheus.yaml [hunk 2]
```diff
@@ -93,6 +93,7 @@ spec:
   listenLocal: true
   nodeSelector:
     beta.kubernetes.io/os: linux
+  tolerations: []
```
Comment (brancz) [inline assets/prometheus-k8s/prometheus.yaml]: I think these are going to get reset by jsonnet generation

Comment (paulfantom) [reply assets/prometheus-k8s/prometheus.yaml]: Yeah, that's why travis was complaining ;)

**Comment by paulfantom** [comment]
/hold

until 4.2

**Comment by brancz** [comment]
nice job!

/lgtm

**Pull request merged**